nservicebus/azure-storage-queues/faq.md

```diff
@@ -15,6 +15,6 @@ tags:
 include: azure-compared-to-msmq
 
 
-### Do Azure Storage Queues provide an exactly once deliver model?
+### Do Azure Storage Queues provide an exactly-once deliver model?
 
-No, it's at least once delivery. To overcome the lack of transactions, these queues work with a so called 'queue-peek-lock' model. When a worker pulls a message from a queue, it will actually become invisible instead of removed from the queue. The worker has to process the messages in a well defined timeframe and delete it explicitly when done. If the worker fails to do this, because it died, an exception was thrown or it was too slow, then the message will reappear on the queue and another instance of the worker can pick it up. This also implies that multiple workers may be working on the same message at the same time (esp. problematic when operations take to long).
\ No newline at end of file
+No, they do not support an exactly-once deliver model. Depending on the selected `TransportTransactionMode` it is either `at-least-once` or `at-most-once` delivery. Azure Storage Queues queues work with a so called Peek-Lock model to overcome the lack of transactions. When a worker pulls a message from a queue, it is marked as invisible for specified period of time. The worker has to process the messages in a well defined timeframe and delete it explicitly when done. When message is not processed within the pre-defined time, a.k.a. Invisibility Timeout, the message will become visible on the queue and another consumer can access it. Depending on a selected `TransportTransactionMode`, the transport deletes a message after processing (`at least once`) or before processing (`at most once`).
```

Comment:
"worker" implies a certain technology on azure. Agree with @dvdstelt, if we want to be closer to the NSB concepts, using "endpoint" would be aligned more.

BTW, it's not a thread, since we run concurrently, could be the same thread 😉